### PR TITLE
[CARBONDATA-3755]Fix clean up issue with respect to segmentMetadaInfo after update and clean files

### PR DESCRIPTION
**Pull request by akashrn5**
### Why is this PR needed?
 1. segmentMetadaInfo is not getting copied to new segment files written after multiple updates and clean files opearation.
2. old segment files are not getting deleted and getting accumulated. 
 
 ### What changes were proposed in this PR?
1. update the segmentMetadaInfo to new files
2. once we write new segment file, delete the old invalid segment files.
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - No
 - Yes

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2569/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/861/

File: core/src/main/java/org/apache/carbondata/core/mutate/CarbonUpdateUtil.java [hunk 1]
```diff
@@ -579,10 +580,28 @@ public static void cleanUpDeltaFiles(CarbonTable table, boolean forceDelete) thr
...
     List<Segment> segmentFilesToBeUpdatedLatest = new ArrayList<>();
+    CarbonFile segmentFilesLocation =
+        FileFactory.getCarbonFile(CarbonTablePath.getSegmentFilesLocation(table.getTablePath()));
     for (Segment segment : segmentFilesToBeUpdated) {
-      String file =
-          SegmentFileStore.writeSegmentFile(table, segment.getSegmentNo(), UUID);
-      segmentFilesToBeUpdatedLatest.add(new Segment(segment.getSegmentNo(), file));
+      SegmentFileStore fileStore =
+          new SegmentFileStore(table.getTablePath(), segment.getSegmentFileName());
+      segment.setSegmentMetaDataInfo(fileStore.getSegmentFile().getSegmentMetaDataInfo());
```
Comment (Indhumathi27) [inline core/src/main/java/org/apache/carbondata/core/mutate/CarbonUpdateUtil.java]: Can you please add a testcase for this issue, if possible

**Comment by ydvpankaj99** [comment]
retest this please

File: core/src/main/java/org/apache/carbondata/core/mutate/CarbonUpdateUtil.java (hunk 1 above)
Comment (akashrn5) [reply core/src/main/java/org/apache/carbondata/core/mutate/CarbonUpdateUtil.java]: added

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/867/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2575/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2579/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/871/

**Comment by kunal642** [comment]
LGTM

**Pull request closed without merging**